src/resolve/resolved-dns-transaction.c

```diff
@@ -2332,11 +2332,19 @@ static int dns_transaction_request_dnssec_rr(DnsTransaction *t, DnsResourceKey *
                 r = dns_transaction_go(aux);
                 if (r < 0)
                         return r;
+                if (ret)
+                        *ret = aux;
```

Comment:
Usually, we unconditionally set `ret` on success. However, this sets `ret` only when this returns `1`.
Please set `NULL` on returning `0`.